docs/development/design/watch-api.md

```diff
@@ -0,0 +1,191 @@
+
+This has the effect of having better bandwith scaling characteristics since only change events are sent to N devices when the desired state changes.  If the devices are not actively joining and leaving the organization, this can be verly little bandwidth.  And even if they are joining and leaving at a constant rate, the bandwidth scales with N (not N x N).  In addition to the bandwidth savings, peer devices are pushed change events as soon as possible for processing which means they can converge on the desired state quicker than waiting for the next polling interval.
+
+### Example
+
+When you use the `watch=true` query parameter the HTTP response body (served as `application/json;stream=watch`) consists of a series of JSON documents.  When you don't specify the the `revision` argument you will get all the devices in an organization like in a typical list requests followed by a `bookmark` event.
+
+1. List all of the devices in a given organization.
+
+   ```console
+   GET /api/organizations/{organization_id}/devices?watch=true
+   ---
+   200 OK
+   Content-Type: application/json
+
+   {
+     "type": "change",
+     "value":  {
+       "user_id": "3833f886-fc40-44df-b4e3-644fa287ed2d",
+       "organization_id": "f13fe84d-9d6e-4dad-8464-bb30e9ef0194",
+       "public_key": "...",
+       "local_ip": "172.17.0.4:58664",
+       "tunnel_ip": "",
+       "child_prefix": null,
+       "relay": false,
+       "discovery": false,
+       "reflexive_ip4": "47.196.141.166",
+       "endpoint_local_address_ip4": "172.17.0.4",
+       "symmetric_nat": true,
+       "hostname": "device2",
+       "revision": 10246
+     }
+   }
+   { "type": "bookmark" }   
+   ```
+
+If a client **watch** operation is disconnected then that client can start a new **watch** from
+the last returned `revision`;
+2. Starting from revision 10245, use the following request to receive notifications of any API operations that affect devices.
+
+   ```console
+   GET /api/organizations/{organization_id}/devices?watch=true&revision=10245
+   ---
+   200 OK
+   Transfer-Encoding: chunked
+   Content-Type: application/json;stream=watch
+
+   { "type": "bookmark" }
+   ...
+   {
+     "type": "delete",
+     "value":  {
+       "user_id": "3833f886-fc40-44df-b4e3-644fa287ed2d",
+       "organization_id": "f13fe84d-9d6e-4dad-8464-bb30e9ef0194",
+       "public_key": "...",
+       "local_ip": "172.17.0.4:58664",
+       "tunnel_ip": "",
+       "child_prefix": null,
+       "relay": false,
+       "discovery": false,
+       "reflexive_ip4": "47.196.141.166",
+       "endpoint_local_address_ip4": "172.17.0.4",
+       "symmetric_nat": true,
+       "hostname": "device2",
+       "revision": 10247
+     }
+   }      
+   ...
+   ```
+
+In the above example, the client gets a bookmark event imediately indicating no changes had occured since revision 10245 but then a little while later a delete event is delivered.
```

Comment:
```suggestion
In the above example, the client gets a bookmark event immediately indicating no changes had occurred since revision 10245, but then a little while later a delete event is delivered.
```